### PR TITLE
Assertion failure in InstructorFeedbackCopyAction: null #4441

### LINKED ISSUE #4441: Assertion failure in InstructorFeedbackCopyAction: null
null

Request Method GET 
User Agent Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/45.0.2454.101 Safari/537.36 
Request Url https://teammatesv4.appspot.com/page/instructorFeedbackCopy 
Request Path /page 
Request Parameters {} 
Stack Trace

java.lang.AssertionError: null
	at teammates.common.util.Assumption.fail(Assumption.java:59)
	at teammates.common.util.Assumption.assertTrue(Assumption.java:27)
	at teammates.common.util.Assumption.assertNotNull(Assumption.java:269)
	at teammates.common.util.Assumption.assertNotNull(Assumption.java:261)
	at teammates.ui.controller.InstructorFeedbackCopyAction.execute(InstructorFeedbackCopyAction.java:29)
	at teammates.ui.controller.Action.executeAndPostProcess(Action.java:337)
	at teammates.ui.controller.ControllerServlet.doPost(ControllerServlet.java:59)
	at teammates.ui.controller.ControllerServlet.doGet(ControllerServlet.java:38)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:617)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1166)
	at teammates.storage.datastore.DatastoreFilter.doFilter(DatastoreFilter.java:28)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.appstats.AppstatsFilter.doFilter(AppstatsFilter.java:142)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.ParseBlobUploadFilter.doFilter(ParseBlobUploadFilter.java:125)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.runtime.jetty.SaveSessionFilter.doFilter(SaveSessionFilter.java:37)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.JdbcMySqlConnectionCleanupFilter.doFilter(JdbcMySqlConnectionCleanupFilter.java:60)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(TransactionCleanupFilter.java:43)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:388)
	at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:765)
	at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:418)
	at com.google.apphosting.runtime.jetty.AppVersionHandlerMap.handle(AppVersionHandlerMap.java:260)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:326)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
	at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:923)
	at com.google.apphosting.runtime.jetty.RpcRequestParser.parseAvailable(RpcRequestParser.java:78)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
	at com.google.apphosting.runtime.jetty.JettyServletEngineAdapter.serviceRequest(JettyServletEngineAdapter.java:147)
	at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.run(JavaRuntime.java:457)
	at com.google.tracing.TraceContext$TraceContextRunnable.runInContext(TraceContext.java:437)
	at com.google.tracing.TraceContext$TraceContextRunnable$1.run(TraceContext.java:444)
	at com.google.tracing.CurrentContext.runInContext(CurrentContext.java:230)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContextNoUnref(TraceContext.java:308)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContext(TraceContext.java:300)
	at com.google.tracing.TraceContext$TraceContextRunnable.run(TraceContext.java:441)
	at com.google.apphosting.runtime.ThreadGroupPool$PoolEntry.run(ThreadGroupPool.java:235)
	at java.lang.Thread.run(Thread.java:745)

### PR DESCRIPTION
**Pull request by kanghj**
Fixes #4441.

Previously, if the user tried to copy but failed, the url will be InstructorFeedbackCopy, but without any parameters. Therefore, If the user then tried to navigate back to this page in future, such as using bookmarks, the error would be thrown.

The first commit causes the test to fail by reloading the failed page.

### CODE REVIEW

File: src/test/java/teammates/test/cases/ui/InstructorFeedbackCopyActionTest.java [hunk 1]
```diff
@@ -101,10 +103,12 @@ public void testExecuteAndPostProcess() throws Exception{
         };
         
         a = getAction(params);
-        ShowPageResult pageResult = (ShowPageResult) a.executeAndPostProcess();
+        RedirectResult pageResult = (RedirectResult) a.executeAndPostProcess();
         
-        assertEquals("/jsp/instructorFeedbacks.jsp?error=true&user=idOfInstructor1OfCourse1",
+        assertEquals("/page/instructorFeedbacksPage?error=true&user=idOfInstructor1OfCourse1",
```
Comment (yj-soh) [inline src/test/java/teammates/test/cases/ui/InstructorFeedbackCopyActionTest.java]: Should these be using `INSTRUCTOR_FEEDBACKS_PAGE` from `Const` instead?

**Comment by thyageshm** [comment]
@kanghj try to complete this soon

**Comment by kanghj** [comment]
@yj-soh, changes made

**Comment by yj-soh** [comment]
Looks okay.
Should we put in a new issue on standardizing between the `Config.getAppUrl().withParam().toString()` pattern and the simple concatenation one over the entire codebase?

**Comment by kanghj** [comment]
@yj-soh, you can go ahead and make an issue for that, if you want.

**Pull request merged**